File: common/src/main/java/com/pluxity/global/config/S3Config.java
package com.pluxity.global.config;

import lombok.Getter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Getter
@Configuration
public class S3Config {

    @Value("${file.s3.bucket}")
    private String bucketName;

    @Value("${file.s3.region}")
    private String region;

    @Value("${file.s3.endpoint-url}")
    private String endpointUrl;

    @Value("${file.s3.access-key}")
    private String accessKey;

    @Value("${file.s3.secret-key}")
    private String secretKey;

    @Value("${file.s3.pre-signed-url-expiration}")
    private int preSignedUrlExpiration;
}

File: common/src/main/java/com/pluxity/global/config/S3Config.kt
package com.pluxity.global.config

import lombok.Getter
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Getter
@Configuration
class S3Config {
    @Value("\${file.s3.bucket}")
    lateinit var bucketName: String

    @Value("\${file.s3.region}")
    lateinit var region: String

    @Value("\${file.s3.endpoint-url}")
    lateinit var endpointUrl: String

    @Value("\${file.s3.access-key}")
    lateinit var accessKey: String

    @Value("\${file.s3.secret-key}")
    lateinit var secretKey: String

    @Value("\${file.s3.pre-signed-url-expiration}")
    val preSignedUrlExpiration = 0
}
